File: test-spring-boot/src/test/java/test/PersonDao.java
package test;

import com.github.afezeria.freedao.annotation.Dao;
import com.github.afezeria.freedao.spring.runtime.DS;

import java.util.List;

/**
 * @author afezeria
 */
@Dao(crudEntity = Person.class)
public interface PersonDao {
    List<Person> queryByIdNotNull();

    int insert(Person person);

    Integer count(Person person);

    @DS(Db.MASTER_1)
    Person selectOneById(Integer id);
}

File: test/src/test/kotlin/test/kotlin/PersonDao.kt
package test.kotlin;

import io.github.afezeria.freedao.annotation.Dao
import test.Person


/**
 * @author afezeria
 */
@Dao(
    crudEntity = Person::class
)
interface PersonDao {
    fun queryByIdGreaterThan(id: Long): List<Person>

    fun list(order: Person?): List<Person>

    fun insert(order: Person): Int

    fun count(order: Person?): Int

    fun selectOneById(id: Long): Person
}
